config/7.1.0/securebanking/ig/scripts/groovy/ProcessRegistration.groovy

```diff
@@ -167,25 +175,32 @@ switch(method.toUpperCase()) {
             return(errorResponse(Status.BAD_REQUEST,"Wrong number of dashes in OI " + organizationalIdentifier +" - expected 2"));
         }
 
-        // Issue: https://github.com/SecureBankingAccessToolkit/securebanking-openbanking-demo/issues/53
-        def dnId = oiComponents[2].toString().replace("Unknown","")
-
-        if (dnId != apiClientOrgCertId) {
-            return(errorResponse(Status.BAD_REQUEST,"apiClientOrg ID in cert " + dnId +" does not match id in SSA " + apiClientOrgCertId));
-
-        }
-
         // TODO: Subject DN for cert bound access tokens
 
-
         // Convert to JSON and pass it on
-
         def regJson = oidcRegistration.build();
-
         logger.debug(SCRIPT_NAME + "final json [" + regJson + "]")
         request.setEntity(regJson)
-        break
 
+        // Verify that the tls transport cert is registered for the TPP's software statement
+        if (apiClientOrgJwksUri != null) {
+            logger.debug(SCRIPT_NAME + "Checking cert against remote jwks: " + apiClientOrgJwksUri)
+            return jwkSetService.getJwkSet(new URL(apiClientOrgJwksUri)).thenAsync(jwkSet -> {
+                return verifyTlsClientCertExistsInJwkSet(jwkSet)
+            }).thenCatchAsync(e -> {
+                logger.debug(SCRIPT_NAME + "failed to get jwks due to exception", e)
+                return newResultPromise(errorResponse(Status.BAD_REQUEST, "unable to get jwks from url: " + apiClientOrgJwksUri))
+            })
+        } else {
+            // Verify against the software_jwks which is a JWKSet embedded within the software_statement
+            // NOTE: this is only suitable for developer testing purposes
+            if (!allowIgIssuedTestCerts) {
+                return(errorResponse(Status.BAD_REQUEST, "software_statement must contain software_jwks_endpoint"));
+            }
+            logger.debug(SCRIPT_NAME + "Checking cert against ssa software_jwks: " + apiClientOrgJwks)
+            def jwkSet = new JWKSet(new JsonValue(apiClientOrgJwks.get("keys")))
+            return verifyTlsClientCertExistsInJwkSet(jwkSet)
+        }
```

Comment:
Extract to a method???

Reply:
As discussed, it's too messy to extract a method without re-writing a lot of the filter. 

Leaving it for now, it would be nice to re-write this in Java, then we can improve the code quality + add unit tests